notebook/log.py

```diff
@@ -6,43 +6,93 @@
 #-----------------------------------------------------------------------------
 
 import json
+
+try:
+    from urllib.parse import urlparse, urlunparse
+except ImportError:
+    # py2
+    from urllib import urlparse, urlunparse
+
 from tornado.log import access_log
+from tornado.web import StaticFileHandler
+
+
+# url params to be scrubbed if seen
+# any url param that *contains* one of these
+# will be scrubbed from logs
+SCRUB_PARAM_KEYS = ('token', 'auth', 'key', 'code', 'state')
+
+
+def _scrub_uri(uri):
+    """scrub auth info from uri"""
+    parsed = urlparse(uri)
+    if parsed.query:
+        # check for potentially sensitive url params
+        # use manual list + split rather than parsing
+        # to minimally perturb original
+        parts = parsed.query.split('&')
+        changed = False
+        for i, s in enumerate(parts):
+            if '=' in s:
+                key, value = s.split('=', 1)
+                for substring in SCRUB_PARAM_KEYS:
+                    if substring in key:
+                        parts[i] = key + '=[secret]'
+                        changed = True
+        if changed:
+            parsed = parsed._replace(query='&'.join(parts))
+            return urlunparse(parsed)
+    return uri
+
+
+def _scrub_headers(headers):
+    """scrub auth info from headers"""
+    headers = dict(headers)
+    if 'Authorization' in headers:
+        auth = headers['Authorization']
+        if auth.startswith('token '):
+            headers['Authorization'] = 'token [secret]'
```

Comment:
Should we scrub any auth header? What about cookies?

Reply:
Scrubbing cookies is a good idea. Will add that.